app/models/listing.py

```diff
@@ -65,34 +65,48 @@ def category_name(self):
     @staticmethod
     def search(**kwargs):
         """ Returns all listings matching the criteria """
-        filter_list = []
         if 'main_search_term' in kwargs:
             term = kwargs['main_search_term']
-            filter_list.append(or_(
-                Listing.name.like('%{}%'.format(term)),
-                Listing.description.like('%{}%'.format(term)))
-            )
+            name = '%{}%'.format(term)
+
         if 'name_search_term' in kwargs:
             term = kwargs['name_search_term']
-            filter_list.append(or_(
-                Vendor.first_name.like('%{}%'.format(term)),
-                Vendor.last_name.like('%{}%'.format(term)),
-                Vendor.company_name.like('%{}%'.format(term)))
-            )
-        if 'available' in kwargs:
-            filter_list.append(Listing.available == kwargs['available'])
+            vendor = '%{}%'.format(term)
 
+        price_min = 0
+        # price_max some arbitrarily large number to ensure all price are shown unless a user specifies a price
+        price_max = float('inf')
+        bookmark_ids = [listing.id for listing in Listing.query.all()]
```

Comment:
It's not the query itself that would slow things down, but that we're always putting all of the listings into memory when we make the list